src/main/java/com/google/api/resourcenames/UntypedResourceName.java

```diff
@@ -78,6 +82,22 @@ public ResourceNameType getType() {
     return UntypedResourceNameType.instance();
   }
 
+  @Override
+  /* Returns a map with an empty String as the sole key, which maps to the raw value of this ResourceName. */
+  public Map<String, String> getFieldValuesMap() {
+    if (fieldValuesMap != null) {
+      return fieldValuesMap;
+    }
+    fieldValuesMap = ImmutableMap.of("", rawValue);
+    return fieldValuesMap;
+  }
+
+  @Override
+  /* Returns the raw value of this ResourceName, regardless of the fieldName parameter. */
```

Comment:
Maybe it should return the raw value if the string is an empty string, and null otherwise? This would make a consistent story with what `getFieldValuesMap()` returns.

Reply:
Sure.